tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl

```diff
@@ -0,0 +1,232 @@
+cwlVersion: v1.1
+class: CommandLineTool
+
+# Extensions
+$namespaces:
+    s: https://schema.org/
+    ilmn-tes: http://platform.illumina.com/rdf/ica/
+$schemas:
+  - https://schema.org/version/latest/schemaorg-current-http.rdf
+
+# Metadata
+s:author:
+    class: s:Person
+    s:name: Sehrish Kanwal
+    s:email: sehrish.kanwal@umccr.org
+
+# ID/Docs
+id: dragen-transcriptome--4.0.3
+label: dragen-transcriptome v(4.0.3)
+doc: |
+    Documentation for dragen-transcriptome v4.0.3
+
+# ILMN V1 Resources Guide: https://illumina.gitbook.io/ica-v1/analysis/a-taskexecution#type-and-size
+# ILMN V2 Resources Guide: https://help.ica.illumina.com/project/p-flow/f-pipelines#compute-types
+# ILMN Resources Guide: https://illumina.gitbook.io/ica-v1/analysis/a-taskexecution#type-and-size
+hints:
+    ResourceRequirement:
+        ilmn-tes:resources:
+            tier: standard
+            type: fpga
+            size: medium
+        coresMin: 16
+        ramMin: 240000
+    DockerRequirement:
+        dockerPull: 699120554104.dkr.ecr.us-east-1.amazonaws.com/public/dragen:4.0.3
+
+requirements:
+  SchemaDefRequirement:
+    types:
+      - $import: ../../../schemas/fastq-list-row/1.0.0/fastq-list-row__1.0.0.yaml
+  InlineJavascriptRequirement:
+    expressionLib:
+      - $include: ../../../typescript-expressions/dragen-tools/4.0.3/dragen-tools__4.0.3.cwljs
+      - $include: ../../../typescript-expressions/utils/1.0.0/utils__1.0.0.cwljs
+    
+  InitialWorkDirRequirement:
+    listing:
+      - entryname: $(get_script_path())
+        entry: |
+          #!/usr/bin/env bash
+
+          # Fail on non-zero exit of subshell
+          set -euo pipefail
+
+          # Initialise dragen
+          /opt/edico/bin/dragen \\
+            --partial-reconfig DNA-MAPPER \\
+            --ignore-version-check true
+
+          # Create directories
+          mkdir --parents \\
+            "$(get_ref_mount())" \\
+            "$(get_intermediate_results_dir())" \\
+            "$(inputs.output_directory)"
+
+          # untar ref data into scratch space
+          tar \\
+            --directory "$(get_ref_mount())" \\
+            --extract \\
+            --file "$(inputs.reference_tar.path)"
+
+          # Run dragen command and import options from cli
+          $(get_dragen_eval_line())
+
+      - |
+        ${
+            return generate_germline_mount_points(inputs);
```

Comment:
If you want, we can create a function called generate_transcriptome_mount_points that calls generate_germline_mount_points underneath?

Reply:
Noted. Think that'll be better - avoids confusion for using germline term in wts (tumor) pipeline.

Reply:
I'll update it.

Reply:
This way you get to write some typescript!